src/sync/sync_manager.cpp

```diff
@@ -387,6 +393,12 @@ std::shared_ptr<SyncSession> SyncManager::get_existing_session_locked(const std:
     return it->second;
 }
 
+std::shared_ptr<SyncSession> SyncManager::get_existing_session(const std::string& path) const
+{
+    std::lock_guard<std::mutex> lock(m_session_mutex);
+    return get_existing_session_locked(path);
```

Comment:
It might. I'm not clear what semantics Java needs though. Can you give me some background on how you intend to use this?